src/library/impl/meta_reader/cache.h

```diff
@@ -149,12 +154,17 @@ namespace xlang::meta::reader
                     }
                 };
 
-                remove(members.interfaces, name);
-                remove(members.classes, name);
-                remove(members.enums, name);
-                remove(members.structs, name);
-                remove(members.delegates, name);
-                remove(members.classes, name);
+                auto pos = members.types.find(name);
+                if (pos != members.types.end())
+                {
+                    members.types.erase(pos);
```

Comment:
I'm not sure we can erase the the type entirely as that would prevent the cache from finding it.

Reply:
Oh, I see, because there may be referenece to that type from type that weren't deleted. So this is really a doc issue (it doesn't remove the types completely, just removes them from the type-specific catalogs).

Reply:
Right.